src/dynamic_boundary_conditions/hirds_depth_data_from_db.py

```diff
@@ -6,75 +6,75 @@
 """
 
 import pandas as pd
-import geopandas
+import pathlib
+import logging
+import sys
+from shapely.geometry import Polygon
 from src.dynamic_boundary_conditions import hirds_depth_data_to_db
+from src.digitaltwin import setup_environment
+from src.dynamic_boundary_conditions import hyetograph
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-def catchment_area_geometry_info(file):
-    """Extract geometry from the shape file, returns polygon."""
-    catchment = geopandas.read_file(file)
-    catchment = catchment.to_crs(4326)
-    catchment_area = catchment.geometry[0]
-    return catchment_area
-
-
-def get_each_site_hirds_depth_data(
-    ari, duration, site, engine, rcp=None, time_period=None
-):
-    """Get hirds rainfall depth data from the database."""
-    if (rcp is None and time_period is not None) or (
-        rcp is not None and time_period is None
-    ):
-        raise ValueError(
-            "check the arguments of get_hirds_depth_data if rcp is None,time"
-            " period should be None and vice-versa"
-        )
+formatter = logging.Formatter("%(levelname)s:%(asctime)s:%(name)s:%(message)s")
+stream_handler = logging.StreamHandler()
+stream_handler.setFormatter(formatter)
+
+log.addHandler(stream_handler)
+
+
+def filter_for_duration(rain_depth: pd.DataFrame, duration: str) -> pd.DataFrame:
+    """Used to filter the hirds rainfall data for a requested duration."""
+    if duration != "all":
+        rain_depth = rain_depth[["site_id", "rcp", "time_period", "ari", "aep", f"{duration}"]]
+    return rain_depth
+
+
+def get_each_site_rain_depth_data(
+        engine, site_id: str, rcp: float, time_period: str, ari: float, duration: str) -> pd.DataFrame:
+    """Get the hirds rainfall data for the requested site from the database and return in dataframe format."""
```

Comment:
It could be good to add some extra documentation to the docstrings for parameters. because I am not aware of what rcp and ari mean since I have never worked on hyetographs before. https://stackoverflow.com/questions/9195455/how-to-document-a-method-with-parameters this link talks about different ways you could do this.

If it is obvious what they mean then you might not need it. @rosepearson what are your thoughts?

Reply:
I have documented them in the README file under the 'Dynamic Boundary Conditions - Get required rainfall data from the database' section. But good thought, I think it would be good to add this parameter information to the relevant functions'  docstrings.

Reply:
Done. I've now added extra documentation to all docstrings for parameters.